3031 ag grid theme dark mode

BluePandaCard opened the pull request:
<!-- 🙏 Thank you for your contribution, it is greatly appreciated! -->

## Summary of the changes
Updates css tokens in ag-theme-icds so that it responds to light/dark mode. Tested in design system for both light and dark mode.

Light mode:

<img width="565" alt="Screenshot 2025-01-16 at 13 52 53" src="https://github.com/user-attachments/assets/6516ba33-a266-41e1-bf21-61d65790c1c6" />

Dark mode:

<img width="569" alt="Screenshot 2025-01-16 at 13 31 50" src="https://github.com/user-attachments/assets/4e90f51c-a0e5-4c83-9d1b-4410eabf27bf" />


## Related issue
#3031 

## Checklist

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

ad9242 commented:
@BluePandaCard apologies, there was some info missing from the ticket. When considering the effort implementing this, we factored in adding a visual regression test in Cypress. Would you be able to do this, please?

BluePandaCard commented:
I've added the visual regression tests. I've added the images that were generated locally for me, but I'm not sure if they'll pass this CI. I tried running the [update Cypress images action](https://github.com/BluePandaCard/ic-ui-kit/actions/runs/12826098513/job/35765367813) on my branch but it fails every other test so if these images fail, it might be another case of someone from the core team having to generate images for me.

GCHQ-Developer-741 commented on packages/react/cypress-image-diff-screenshots/baseline/AgGridWrapper.cy.tsx-ag-grid-light.png:
Do you think the extra border around the car details is from generating them locally?

BluePandaCard replied:
Potentially because that doesn't happen when clicking on car details on the website example

GCHQ-Developer-847 replied:
Yeah had a look when running the Cypress tests - that border is a focus indicator and for some reason the car details automatically gets focused

GCHQ-Developer-847 commented:
> I've added the visual regression tests. I've added the images that were generated locally for me, but I'm not sure if they'll pass this CI. I tried running the [update Cypress images action](https://github.com/BluePandaCard/ic-ui-kit/actions/runs/12826098513/job/35765367813) on my branch but it fails every other test so if these images fail, it might be another case of someone from the core team having to generate images for me.

Looks like they are failing - I will sort them
